tests/components/homeassistant_green/test_init.py

```diff
@@ -0,0 +1,75 @@
+"""Test the Home Assistant Green integration."""
+from unittest.mock import patch
+
+from homeassistant.components.homeassistant_green.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, MockModule, mock_integration
+
+
+async def test_setup_entry(hass: HomeAssistant) -> None:
+    """Test setup of a config entry."""
+    mock_integration(hass, MockModule("hassio"))
+
+    # Setup the config entry
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={},
+        title="Home Assistant Green",
+    )
+    config_entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.homeassistant_green.get_os_info",
+        return_value={"board": "green"},
+    ) as mock_get_os_info:
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert len(mock_get_os_info.mock_calls) == 1
+
+    # Test unloading the config entry
+    assert await hass.config_entries.async_unload(config_entry.entry_id)
+
+
+async def test_setup_entry_wrong_board(hass: HomeAssistant) -> None:
+    """Test setup of a config entry with wrong board type."""
+    mock_integration(hass, MockModule("hassio"))
+
+    # Setup the config entry
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={},
+        title="Home Assistant Green",
+    )
+    config_entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.homeassistant_green.get_os_info",
+        return_value={"board": "generic-x86-64"},
+    ) as mock_get_os_info:
+        assert not await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert len(mock_get_os_info.mock_calls) == 1
```

Comment:
Should we also assert that the entry was removed?

Reply:
Fixed by https://github.com/home-assistant/core/pull/99361